script/LookupEditor.js

```diff
@@ -63,25 +69,35 @@ var LookupEditor = function ($table) {
      */
     function addForm(data) {
         if ($form) $form.remove();
+        var $agg = $table.parents('.structaggregation');
+
         $form = jQuery('<form></form>');
         $form.html(data);
-        $table.parents('.structaggregation').append($form);
-
+        jQuery('<input>').attr({
+            type: 'hidden',
+            name: 'searchconf',
+            value: $agg.attr('data-searchconf')
+        }).appendTo($form); // add the search config to the form
+        $agg.append($form);
         EntryEditor($form);
 
+        var $errors = $form.find('div.err').hide();
+
         $form.submit(function (e) {
             e.preventDefault();
+            $errors.hide();
 
             jQuery.post(
                 DOKU_BASE + 'lib/exe/ajax.php',
                 $form.serialize()
             )
-                .done(function () {
-                    // FIXME
+                .done(function (data) {
+                    $table.find('tbody').append(data);
+                    addDeleteRowButtons(); // add the delete button to the new row
+                    addForm(formdata); // reset the whole form
                 })
                 .fail(function (xhr) {
-                    // FIXME
-                    alert(xhr.responseText)
+                    $errors.text(xhr.responseText).show();
```

Comment:
See https://github.com/cosmocode/dokuwiki-plugin-struct/blob/lookups/action/lookup.php#L69 I send a text/plain response with the exception text.